### PR TITLE
Add support for clean pod policy all

### PR DESCRIPTION
**Pull request by ChenYi015**
## Proposed Changes

- Add support for clean pod policy all

### CODE REVIEW

File: pkg/controller.v1/tensorflow/job.go [hunk 1]
```diff
@@ -175,22 +175,25 @@ func (tc *TFController) deletePodsAndServices(tfJob *tfv1.TFJob, pods []*v1.Pod)
 		return nil
 	}
 
-	// Delete nothing when the cleanPodPolicy is None.
-	if *tfJob.Spec.CleanPodPolicy == common.CleanPodPolicyNone {
-		return nil
+	cleanPodPolicy := common.CleanPodPolicyUndefined
```
Comment (cheyang) [inline pkg/controller.v1/tensorflow/job.go]: I suggest encapsulating the logic into a method to improve reusability and clarity.  Such as

```golang
func getCleanPodPolicy(tfJob *tfv1.TFJob) common.CleanPodPolicy {
	if tfJob.Spec.CleanPodPolicy != nil {
		return *tfJob.Spec.CleanPodPolicy
	}
	return common.CleanPodPolicyUndefined
}
```

Comment (ChenYi015) [reply pkg/controller.v1/tensorflow/job.go]: @cheyang Thanks for the review. Done!

File: pkg/controller.v1/tensorflow/job.go [hunk 2]
```diff
@@ -175,22 +175,21 @@ func (tc *TFController) deletePodsAndServices(tfJob *tfv1.TFJob, pods []*v1.Pod)
 		return nil
 	}
 
-	// Delete nothing when the cleanPodPolicy is None.
-	if *tfJob.Spec.CleanPodPolicy == common.CleanPodPolicyNone {
+	cleanPodPolicy := getCleanPodPolicy(tfJob)
+	switch cleanPodPolicy {
+	case common.CleanPodPolicyUndefined, common.CleanPodPolicyNone:
+		// Do nothing when the cleanPodPolicy is undefined or none.
```
Comment (happy2048) [inline pkg/controller.v1/tensorflow/job.go]: 这里加一行日志吧，当没有设置时，行为是None

Comment (ChenYi015) [reply pkg/controller.v1/tensorflow/job.go]: Done.

File: pkg/controller.v1/tensorflow/job.go [hunk 3]
```diff
@@ -6,7 +6,7 @@ import (
 	"time"
 
 	log "github.com/sirupsen/logrus"
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
```
Comment (cheyang) [inline pkg/controller.v1/tensorflow/job.go]: How about using `corev1` to replace `v1`.

File: pkg/controller.v1/tensorflow/job.go [hunk 4]
```diff
@@ -175,22 +175,25 @@ func (tc *TFController) deletePodsAndServices(tfJob *tfv1.TFJob, pods []*v1.Pod)
...
 
-	for _, pod := range pods {
-		if *tfJob.Spec.CleanPodPolicy == common.CleanPodPolicyRunning && pod.Status.Phase != v1.PodRunning && pod.Status.Phase != v1.PodPending { // && pod.Status.Phase != v1.PodUnknown
-			continue
-		}
-		if err := tc.PodControl.DeletePod(pod.Namespace, pod.Name, tfJob); err != nil {
+	cleanPodPolicy := getCleanPodPolicy(tfJob)
+	switch cleanPodPolicy {
+	case common.CleanPodPolicyUndefined, common.CleanPodPolicyNone:
+		logger.Debugf("Do nothing when the clean pod policy is %s.", cleanPodPolicy)
```
Comment (cheyang) [inline pkg/controller.v1/tensorflow/job.go]: I suggest using `info` level here.

File: pkg/controller.v1/tensorflow/job.go (hunk 3 above)
Comment (ChenYi015) [reply pkg/controller.v1/tensorflow/job.go]: Done.

File: pkg/controller.v1/tensorflow/job.go (hunk 4 above)
Comment (ChenYi015) [reply pkg/controller.v1/tensorflow/job.go]: Done.

**Review by cheyang — APPROVED** [review]
/lgtm
/approve

**Pull request merged**